tmt/utils.py

```diff
@@ -3628,3 +3698,136 @@ def __init__(
+        default_factory: Callable[[], T],
+        # Options
+        option: Optional[FieldCLIOption] = None,
+        is_flag: bool = False,
+        choices: Union[None, Sequence[str], Callable[[], Sequence[str]]] = None,
+        multiple: bool = False,
+        metavar: Optional[str] = None,
+        help: Optional[str] = None,
+        # Input data normalization
+        normalize: Optional[NormalizeCallback[T]] = None
+        ) -> T:
+    pass
+
+
+# TODO: ignore[misc]: mypy reports implementation cannot handle all possible arguments of
+# signatures 2 and 3, but I can't find the issue :/
+def field(  # type: ignore[misc]
+        *,
+        default: Any = dataclasses.MISSING,
+        default_factory: Any = dataclasses.MISSING,
+        # Options
+        option: Optional[FieldCLIOption] = None,
+        is_flag: bool = False,
+        choices: Union[None, Sequence[str], Callable[[], Sequence[str]]] = None,
+        multiple: bool = False,
+        metavar: Optional[str] = None,
+        help: Optional[str] = None,
+        # Input data normalization
+        normalize: Optional[NormalizeCallback[T]] = None
+        ) -> Any:
+    """
+    Define a :py:class:`DataContainer` field.
+
+    Effectively a fancy wrapper over :py:func:`dataclasses.field`, tailored for
+    tmt code needs and simplification of various common tasks.
+
+    :param default: if provided, this will be the default value for this field.
+        Passed directly to :py:func:`dataclass.field`.
+        It is an error to specify both ``default`` and ``default_factory``.
+    :param default_factory: if provided, it must be a zero-argument callable
+        that will be called when a default value is needed for this field.
+        Passed directly to :py:func:`dataclass.field`.
+        It is an error to specify both ``default`` and ``default_factory``.
+    :param option: one or more command-line option names.
+        Passed directly to :py:func:`click.option`.
+    :param is_flag: marks this option as a flag.
+        Passed directly to :py:func:`click.option`.
+    :param choices: if provided, the command-line option would accept only
+        the listed input values.
+        Passed to :py:func:`click.option` as a :py:class:`click.Choice` instance.
+    :param metavar: how the input value is represented in the help page.
+        Passed directly to :py:func:`click.option`.
+    :param help: the help string for the command-line option.
+        Passed directly to :py:func:`click.option`.
+    :param normalize: a callback for normalizing the input value. Consumed by
+        :py:class:`NormalizeKeysMixin`.
+    """
+
+    if default is dataclasses.MISSING and default_factory is dataclasses.MISSING:
+        raise GeneralError('Container field must define one of default or default_factory')
+
+    metadata: FieldMetadata[T] = FieldMetadata()
+
+    if option:
+        assert is_flag is False or isinstance(default, bool)
+
+        metadata.option_args = option
```

Comment:
I think so. I'd convert the single option to a single-item tuple, to keep types simpler. But I agree, as you showed above, a single option as a string should be accepted by the `field()` function.

Reply:
Actually, it's not needed. Options are already defined as `FieldCLIOption` which is a `Union[str, Sequence[str]]`, and `metadata.option_args` are passed to Click and the type is just what Click expects, and since we have no other use for this field, I'd keep the simple `=`, with no conversions involved.

Reply:
Nevermind, it must be done, later we do `*metadata.option_args`, which doesn't work with a string.

Reply:
Fixed in 27a2f60c64fba87236ee3cec83b26f1b8bb64da0

Reply:
Yeap, unpacking was the motivation of my comment. Thanks for the quick fix!

Reply:
Yeah, I'm deep in commands and scripts, and I didn't switch the context fast enough.